static/js/components/google_map.tsx

```diff
@@ -0,0 +1,226 @@
+        strokeWeight: 1,
+        fillOpacity: 0.15,
+      });
+      polygon.setMap(map);
+    }
+  }
+}
+
+/**
+ * Draw a map marker at a specific lat/long.
+ * @param location lat/long position of marker
+ * @param map map to draw marker on
+ */
+function drawMarker(
+  location: GoogleMapCoordinates,
+  map: google.maps.Map
+): void {
+  map.setCenter(location);
+  const marker = new google.maps.Marker({
+    position: location,
+  });
+  marker.setMap(map);
+}
+
+export class GoogleMap extends React.Component<
+  GoogleMapPropType,
+  GoogleMapStateType
+> {
+  div: React.RefObject<HTMLDivElement>;
+
+  constructor(props: GoogleMapPropType) {
+    super(props);
+    this.div = React.createRef();
+    this.state = {
+      markerLocation: { lat: null, lng: null },
+      mapInfo: {
+        up: null,
+        down: null,
+        left: null,
+        right: null,
+        coordinateSequenceSet: [],
+      },
+      shouldShowMap: false,
+    };
+  }
+
+  render(): JSX.Element {
+    return (
+      <>
+        {this.state.shouldShowMap && (
+          <div className="map-container" ref={this.div}></div>
+        )}
+      </>
+    );
+  }
+
+  componentDidMount(): void {
+    this.fetchData();
+  }
+
+  componentDidUpdate(): void {
+    if (this.state.shouldShowMap) {
+      // initialize Map
+      const map = initMap(this.div.current);
+
+      // plot KML Coordinates
+      if (Object.values(this.state.mapInfo).every((val) => val !== null)) {
+        drawKmlCoordinates(this.state.mapInfo, map);
+      }
+      // plot marker
+      else if (this.state.markerLocation.lat && this.state.markerLocation.lng) {
+        drawMarker(this.state.markerLocation, map);
+      }
+    }
+  }
+
+  private fetchData(): void {
+    // Get KML Coordinates for polygons
+    const polygonPromise = axios
+      .get(`/api/place/mapinfo/${this.props.dcid}`)
+      .then((resp) => resp.data)
+      .catch((error) => console.log(error));
+
+    // Get lat/long from properties
+    const latitudePromise = axios
+      .get(`/api/browser/propvals/latitude/${this.props.dcid}`)
+      .then((resp) => resp.data)
+      .catch((error) => console.log(error));
+    const longitudePromise = axios
+      .get(`/api/browser/propvals/longitude/${this.props.dcid}`)
+      .then((resp) => resp.data)
+      .catch((error) => console.log(error));
+
+    Promise.all([polygonPromise, latitudePromise, longitudePromise])
+      .then(([mapInfo, latitudes, longitudes]) => {
+        // Update state with mapInfo
+        if (!_.isEmpty(mapInfo)) {
+          this.setState({
+            mapInfo: mapInfo,
+            shouldShowMap: true,
+          });
+        }
+
+        // Update state with lat/long if both are present
+        if (
+          latitudes.values &&
+          longitudes.values &&
+          !_.isEmpty(latitudes.values.out) &&
+          !_.isEmpty(longitudes.values.out)
+        ) {
+          //TODO (juliawu): Update logic to use highest precision lat/long if
```

Comment:
super nit: always a space after `//`. please see style guide for commenting format (e.g following line starts from after `// `.

Reply:
Fixed!